File: modules/jooby-apitool/src/test/java/parser/UsePathApp.java
package parser;

import org.jooby.Jooby;

import java.util.ArrayList;
import java.util.List;

public class UsePathApp extends Jooby {

  {

    /** /api/path */
    path("/api/path", () -> {
      /** GET /api/path */
      get(() -> {
        List<Foo> foos = new ArrayList<>();
        return foos;
      });
      /** GET /api/path/:id */
      get("/:id", req -> {
        int id = req.param("id").intValue();
        return new Foo(id);
      });
    });
  }
}

File: modules/jooby-apitool/src/test/java/kt/UsePathApp.kt
package kt

import org.jooby.*
import parser.*

class UsePathApp : Kooby({

  /**
   * Summary API.
   */
  path("/api/path") {
    /** List all. */
    get({->
      val foos = listOf<Foo>()
      foos
    })

    /**
     * List one.
     * @param id ID.
     * @return Foo.
     */
    get("/:id") {->
      val id = param("id").intValue
      Foo(id)
    }
  }

})
